RELEASE/scripts/autoscend/iotms/mr2022.ash

```diff
@@ -402,4 +419,140 @@ boolean auto_handleParka()
+	{
+		equip($item[continuum transfunctioner]);
+	}
+
+	foreach index,loc in get_autumnaton_locations()
+	{
+		if(loc == canAdventureInloc)
+		{
+			return true;
+		}
+	}
+	return false;
+}
+
+boolean auto_autumnatonReadyToQuest()
+{
+	if(!auto_hasAutumnaton())
+	{
+		return false;
+	}
+
+	return item_amount($item[autumn-aton]) != 0;
+}
+
+location auto_autumnatonQuestingIn()
+{
+	return to_location(get_property("autumnatonQuestLocation"));
+}
+
+boolean auto_autumnatonCheckForUpgrade(string upgrade)
+{
+	string currentUpgrades = get_property("autumnatonUpgrades");
+	if(contains_text(currentUpgrades,upgrade))
+	{
+		return true;
+	}
+	return false;
+}
+
+boolean auto_sendAutumnaton(location loc)
+{
+	if(auto_autumnatonCanAdv(loc))
+	{
+		cli_execute("autumnaton send " + loc);
+		handleTracker("Autumnaton sent to " + loc, "auto_otherstuff");
+		return true;
+	}
+	return false;
+}
+
+void auto_autumnatonQuest()
+{
+	if(!auto_autumnatonReadyToQuest()) return;
+
+	// complete any pending upgrades if it just returned
+	if (total_turns_played() == get_property("autumnatonQuestTurn").to_int() + 1)
+	{
+		catch cli_execute("autumnaton upgrade");
+	}
+
+	// prioritize getting important upgrades
+	if(!auto_autumnatonCheckForUpgrade("leftarm1"))
+	{
+		if(auto_sendAutumnaton($location[The Haunted Pantry]))
+		{
+			return;
+		}
+		else
+		{
+			abort("Haunted pantry should always be available for autumnaton, but autoscend determined it is not. Report issue.");
+		}
+	}
+
+	if(!auto_autumnatonCheckForUpgrade("leftleg1"))
+	{
+		// some bat zones may not be adventured in, so try them all
+		if(auto_sendAutumnaton($location[Guano Junction])) return;
+		if(auto_sendAutumnaton($location[The Batrat And Ratbat Burrow])) return;
+		if(auto_sendAutumnaton($location[The Beanbat Chamber])) return;
+	}
+
+	if(!auto_autumnatonCheckForUpgrade("rightleg1"))
+	{
+		if(auto_sendAutumnaton($location[The Haunted Library])) return;
+	}
+
+	// acquire items to help quests
+	if(fastenerCount() < 30 && lumberCount() < 30)
+	{
+		if(auto_sendAutumnaton($location[The Smut Orc Logging Camp])) return;
+	}
+
+	// should we go regardless of if we have arm upgrades?
+	if(auto_autumnatonCheckForUpgrade("leftarm1") &&
+	 auto_autumnatonCheckForUpgrade("rightarm1") &&
+	 item_amount($item[barrel of gunpowder]) < 5 && 
+	 get_property("sidequestLighthouseCompleted") == "none")
+	{
+		if(auto_sendAutumnaton($location[Sonofa Beach])) return;
+	}
+
+	if(hedgeTrimmersNeeded() > 0)
+	{
+		if(auto_sendAutumnaton($location[Twin Peak])) return;
+	}
+
+	if(!contains_text(get_property("nsTowerDoorKeysUsed"), "digital key") &&
+	item_amount($item[Digital Key]) == 0 &&
+	creatable_amount($item[Digital Key]) == 0 &&
+	!auto_hasPowerfulGlove())
+	{
+		if(!possessEquipment($item[continuum transfunctioner]) && my_level() >= 2)
+		{
+			// unlock 8-bit zone
+			woods_questStart();
+		}
+		if(auto_sendAutumnaton($location[8-bit realm])) return;
+	}
+
+	return;
```

Comment:
do we also want to send to twin peak at any point for chance of getting hedge clippers?

Reply:
I had that originally but took it out because the camel spit makes it trivial. Good point though that it probably is worth doing for folks that don't have the camel. Added

Reply:
Hmm well nevermind I must be thinking of something else because still have code to send to twin peak at line 539. I'll add a condition to ensure we don't have camel when sending to twin peaks